spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigUtils.java

```diff
@@ -37,6 +40,18 @@
 	private ConfigUtils() {
 	}
 
+	public static Map<String, Object> withPrefix(Map<String, Object> data, String prefix, String profile) {
```

Comment:
one of the core changes, how the prefix is computed, for both profile and non-profile based sources.